tests/test_schemadef_plugin.py

```diff
@@ -0,0 +1,77 @@
+#
+# Copyright 2017 Pixar Animation Studios
+#
+# Licensed under the Apache License, Version 2.0 (the "Apache License")
+# with the following modification; you may not use this file except in
+# compliance with the Apache License and the following modification to it:
+# Section 6. Trademarks. is deleted and replaced with:
+#
+# 6. Trademarks. This License does not grant permission to use the trade
+#    names, trademarks, service marks, or product names of the Licensor
+#    and its affiliates, except as required to comply with Section 4(c) of
+#    the License and to reproduce the content of the NOTICE file.
+#
+# You may obtain a copy of the Apache License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the Apache License with the above modification is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied. See the Apache License for the specific
+# language governing permissions and limitations under the Apache License.
+#
+import unittest
+import os
+
+import opentimelineio as otio
+from tests import baseline_reader
+
+"""Unit tests for the schemadef plugin system."""
+
+
+SCHEMADEF_NAME = "schemadef_example"
+EXAMPLE_ARG = "exampleArg"
+EXCLASS = "<class 'opentimelineio.schemadef.example_schemadef.exampleSchemaDef'>"
+
+
+class TestPluginSchemadefs(unittest.TestCase):
+    def setUp(self):
+        self.save_manifest = otio.plugins.manifest._MANIFEST
+        self.save_manifest_path = os.environ.get('OTIO_PLUGIN_MANIFEST_PATH')
+        # find the path to the baselines/schemadef_example.json
+        self.manifest_path = baseline_reader.path_to_baseline(SCHEMADEF_NAME)
+        os.environ['OTIO_PLUGIN_MANIFEST_PATH'] = self.manifest_path
+        otio.plugins.manifest.ActiveManifest(force_reload=True)
+
+    def tearDown(self):
+        # restore original state
+        if self.save_manifest_path:
+            os.environ['OTIO_PLUGIN_MANIFEST_PATH'] = self.save_manifest_path
+        else:
+            del os.environ['OTIO_PLUGIN_MANIFEST_PATH']
+        otio.plugins.manifest._MANIFEST = self.save_manifest
+
+    def test_plugin_schemadef(self):
+        # Our test manifest should have been loaded, including
+        # the example_schemadef.
+        peculiar_value = "something One-derful"
+        try:
```

Comment:
You can use the `with assertRaises` syntax to do this, see:
https://github.com/PixarAnimationStudios/OpenTimelineIO/blob/f5a4480d7418cfbfea6b7604e9d28b46e4c683aa/tests/test_opentime.py#L88

Reply:
The intent here was to check that it DOESN'T raise NotSupportedError, whereas self.assertRaises checks that it DOES raise the exception, I think.  But this condition likely could be removed anyway, since I took out the code in instance_from_schema that could raise this exception.  Instead now it will return an UnknownSchema object, so I should check for that instead, I guess.

Reply:
Oh, and it already does check that by verifying the type of the created class, so if it is UnknownSchema rather than exampleSchemaDef, the type check will fail.